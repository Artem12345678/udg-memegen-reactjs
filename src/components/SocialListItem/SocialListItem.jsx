import React from "react";
import PropTypes from "prop-types";

import instagram from "../../assets/instagram.svg";
import github from "../../assets/github.svg";

import "./SocialListItem.scss";

const SocialListItem = ({ url, icon }) => {
  const image = icon === "instagram" ? instagram : github;

  return (
    <a
      className="social__link"
      href={url}
      target="_blank"
      rel="noopener noreferrer"
    >
      <img src={image} alt="socialIcon" />
    </a>
  );
};

SocialListItem.propTypes = {
  url: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired
};

export default SocialListItem;
