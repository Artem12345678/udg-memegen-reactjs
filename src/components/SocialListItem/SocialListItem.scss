.social__link {
  display: flex;
  width: 28px;
  height: 28px;
  transition: border-color 0.25s ease-in-out;
  text-align: center;
  border: 1px solid transparent;
  border-radius: 50%;
  align-items: center;
  justify-content: center;

  &:focus,
  &:active,
  &:hover {
    border-color: #fff;
  }
}
