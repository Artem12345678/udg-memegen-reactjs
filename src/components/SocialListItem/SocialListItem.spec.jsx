import React from "react";
import { shallow } from "enzyme";

import SocialListItem from "./SocialListItem";

import instagram from "../../assets/instagram.svg";
import github from "../../assets/github.svg";

describe("SocialListItem.jsx", () => {
  let wrapper;

  const links = [
    {
      url: "https://www.github.com/Artem12345678",
      icon: "github"
    },
    {
      url: "https://www.instagram.com/ystimenkoartem",
      icon: "instagram"
    }
  ];

  it("matches snapshot", () => {
    wrapper = shallow(
      <SocialListItem url={links[0].url} icon={links[0].icon} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Github icon", () => {
    wrapper = shallow(
      <SocialListItem url={links[0].url} icon={links[0].icon} />
    );

    expect(wrapper.find(".social__link").exists()).toBe(true);
    expect(
      wrapper
        .find(".social__link")
        .at(0)
        .props().href
    ).toBe(links[0].url);
    expect(
      wrapper
        .find(".social__link img")
        .at(0)
        .props().src
    ).toBe(github);
  });

  it("renders Instagram icon", () => {
    wrapper = shallow(
      <SocialListItem url={links[1].url} icon={links[1].icon} />
    );

    expect(wrapper.find(".social__link").exists()).toBe(true);
    expect(
      wrapper
        .find(".social__link")
        .at(0)
        .props().href
    ).toBe(links[1].url);
    expect(
      wrapper
        .find(".social__link img")
        .at(0)
        .props().src
    ).toBe(instagram);
  });
});
