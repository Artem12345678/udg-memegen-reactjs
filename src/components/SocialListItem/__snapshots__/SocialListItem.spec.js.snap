// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialListItem.jsx matches snapshot 1`] = `
<a
  className="social__link"
  href="https://www.github.com/Artem12345678"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="socialIcon"
    src="github.svg"
  />
</a>
`;
